components/BasicItemSlot.jsx

```diff
@@ -0,0 +1,90 @@
+import styled from "styled-components";
+import basicItemsList from "../data/basicitems-data.json";
+import slugify from "slugify";
+
+export const BasicItemSlot = ({ register, index, watch, basicItemRemove }) => {
+  const value = watch(`basicItemSlotArray.${index}.selected-item`);
+  const selectedItem = basicItemsList.find(
+    (basicItem) => slugify(basicItem.name) === value
```

Comment:
You're using slugify during the runtime. I think it'd make sense to pre-compute all slugs and hard-code them in your json files.